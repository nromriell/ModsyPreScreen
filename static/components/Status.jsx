import React, {Fragment} from 'react'
import {connect} from 'react-redux'

const Status = (props) => {
    return <Fragment>
        { props.loading && <div className="Loading">Loading...</div> }
        { props.error && <div className="Error">Error Reaching Server</div> }
    </Fragment>
};

const mapStateToProps = (state) =>
{
    return {
        loading: state.data.loading,
        error: state.data.error
    }
};

export default connect(mapStateToProps, null)(Status);