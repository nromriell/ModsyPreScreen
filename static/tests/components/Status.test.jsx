import React from 'react'
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import Status from '../../components/Status'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([]);

describe('Status Tests', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            data: {loading: false, error:false, products:[]},
            scrollTarget: 500
        });
        component = renderer.create(
            <Provider store={store}>
                <Status/>
            </Provider>
        )
    });

    it('Renders Correctly', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

});