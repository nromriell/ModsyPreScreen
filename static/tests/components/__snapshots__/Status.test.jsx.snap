// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Tests Renders Correctly 1`] = `null`;
